GitCommands/Git/GitModule.cs

```diff
@@ -1343,6 +1343,8 @@ public string GetCurrentSubmoduleLocalPath()
                 return null;
             }
 
+            Debug.Assert(WorkingDir.StartsWith(SuperprojectModule.WorkingDir), "Submodule working dir should start with super-project's working dir");
```

Comment:
A submodule can be relative the supermodule. (The superproject could still be prefix though)

Reply:
Happy to remove this assert, but are you sure it's invalid?

Look at the next line:

```c#
WorkingDir.Substring(SuperprojectModule.WorkingDir.Length)
```

It seems to be snipping the `SuperprojectModule.WorkingDir` from the current `WorkingDir`, presumably under the assumption that the submodule is nested below the super project.

Reply:
It seems like GE is not supporting this properly then.
An assert is OK as if you build with Debug, you can rebuild.

Just add a comment

Reply:
Sounds like we should open an issue about this.

Would the necessary condition be that a submodule exists _above_ the supermodule in the file system? I've never tried that as I assumed it wouldn't be possible. I'm not sure how to describe the bug  for an issue's description.

Reply:
I can open an issue, should test a little first
(Have never used this feature myself - needed it once but then I was told it was not possible.)